### ISSUE TITLE
Control_transfer returned error, index 768 from D435i

### ISSUE THREAD

**Issue by MiPlayer123** [realsenseai/realsense-ros#1957]
Hello

I am trying to use the D435i on my Jetson Nano with ROS, but I am getting this error: `(messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61` when I run this command `roslaunch realsense2_camera rs_camera.launch`

None of the topics are being published and I cannot see anything in Rviz. If I use the RealSense viewer, everything works fine. The camera is connected to the USB3 port on the Jetson Nano. I have looked at other similar forum posts, but now of them seemed to fix the issue. Any help would be appreciated. 

Thanks

**Comment by MartyG-RealSense** [comment]
Hi @MiPlayer123    If you are receiving the **control_transfer returned** warning only a small number of times during launch and then they stop, or they occur one or two times a minute after launch, then this warning can be safely ignored.  If this warning is generating continuously though then it can indicate a serious communication problem with the device.

The first test that I recommend when trying to correct this issue is to add an **initial_reset:=true** command to the roslaunch instruction to reset the camera at launch:

**roslaunch realsense2_camera rs_camera.launch initial_reset:=true**

Does using this command in your roslaunch instruction make a positive difference to the performance of your launch, please?

**Comment by MiPlayer123** [comment]
@MartyG-RealSense Thanks for your reply. Resetting the camera does not seem to help. I am still getting the same error. I also noticed that very few of the topics are being published properly. I cannot see a video feed in Rviz or even the Odom topic. I am not sure where the error is coming from.

**Comment by MartyG-RealSense** [comment]
Could you paste the text of the terminal log from your roslaunch into a comment please?  Thanks very much.

**Comment by MiPlayer123** [comment]
Here is the log:
```
SUMMARY
========

PARAMETERS
 * /camera/realsense2_camera/accel_fps: -1
 * /camera/realsense2_camera/accel_frame_id: camera_accel_frame
 * /camera/realsense2_camera/accel_optical_frame_id: camera_accel_opti...
 * /camera/realsense2_camera/align_depth: False
 * /camera/realsense2_camera/aligned_depth_to_color_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/allow_no_texture_points: False
 * /camera/realsense2_camera/base_frame_id: camera_link
 * /camera/realsense2_camera/calib_odom_file: 
 * /camera/realsense2_camera/clip_distance: -2.0
 * /camera/realsense2_camera/color_fps: -1
 * /camera/realsense2_camera/color_frame_id: camera_color_frame
 * /camera/realsense2_camera/color_height: -1
 * /camera/realsense2_camera/color_optical_frame_id: camera_color_opti...
 * /camera/realsense2_camera/color_width: -1
 * /camera/realsense2_camera/confidence_fps: -1
 * /camera/realsense2_camera/confidence_height: -1
 * /camera/realsense2_camera/confidence_width: -1
 * /camera/realsense2_camera/depth_fps: -1
 * /camera/realsense2_camera/depth_frame_id: camera_depth_frame
 * /camera/realsense2_camera/depth_height: -1
 * /camera/realsense2_camera/depth_optical_frame_id: camera_depth_opti...
 * /camera/realsense2_camera/depth_width: -1
 * /camera/realsense2_camera/device_type: 
 * /camera/realsense2_camera/enable_accel: False
 * /camera/realsense2_camera/enable_color: True
 * /camera/realsense2_camera/enable_confidence: True
 * /camera/realsense2_camera/enable_depth: True
 * /camera/realsense2_camera/enable_fisheye1: False
 * /camera/realsense2_camera/enable_fisheye2: False
 * /camera/realsense2_camera/enable_fisheye: False
 * /camera/realsense2_camera/enable_gyro: False
 * /camera/realsense2_camera/enable_infra1: False
 * /camera/realsense2_camera/enable_infra2: False
 * /camera/realsense2_camera/enable_infra: False
 * /camera/realsense2_camera/enable_pointcloud: False
 * /camera/realsense2_camera/enable_pose: False
 * /camera/realsense2_camera/enable_sync: False
 * /camera/realsense2_camera/filters: 
 * /camera/realsense2_camera/fisheye1_frame_id: camera_fisheye1_f...
 * /camera/realsense2_camera/fisheye1_optical_frame_id: camera_fisheye1_o...
 * /camera/realsense2_camera/fisheye2_frame_id: camera_fisheye2_f...
 * /camera/realsense2_camera/fisheye2_optical_frame_id: camera_fisheye2_o...
 * /camera/realsense2_camera/fisheye_fps: -1
 * /camera/realsense2_camera/fisheye_frame_id: camera_fisheye_frame
 * /camera/realsense2_camera/fisheye_height: -1
 * /camera/realsense2_camera/fisheye_optical_frame_id: camera_fisheye_op...
 * /camera/realsense2_camera/fisheye_width: -1
 * /camera/realsense2_camera/gyro_fps: -1
 * /camera/realsense2_camera/gyro_frame_id: camera_gyro_frame
 * /camera/realsense2_camera/gyro_optical_frame_id: camera_gyro_optic...
 * /camera/realsense2_camera/imu_optical_frame_id: camera_imu_optica...
 * /camera/realsense2_camera/infra1_frame_id: camera_infra1_frame
 * /camera/realsense2_camera/infra1_optical_frame_id: camera_infra1_opt...
 * /camera/realsense2_camera/infra2_frame_id: camera_infra2_frame
 * /camera/realsense2_camera/infra2_optical_frame_id: camera_infra2_opt...
 * /camera/realsense2_camera/infra_fps: 30
 * /camera/realsense2_camera/infra_height: 480
 * /camera/realsense2_camera/infra_rgb: False
 * /camera/realsense2_camera/infra_width: 848
 * /camera/realsense2_camera/initial_reset: True
 * /camera/realsense2_camera/json_file_path: 
 * /camera/realsense2_camera/linear_accel_cov: 0.01
 * /camera/realsense2_camera/odom_frame_id: camera_odom_frame
 * /camera/realsense2_camera/ordered_pc: False
 * /camera/realsense2_camera/pointcloud_texture_index: 0
 * /camera/realsense2_camera/pointcloud_texture_stream: RS2_STREAM_COLOR
 * /camera/realsense2_camera/pose_frame_id: camera_pose_frame
 * /camera/realsense2_camera/pose_optical_frame_id: camera_pose_optic...
 * /camera/realsense2_camera/publish_odom_tf: True
 * /camera/realsense2_camera/publish_tf: True
 * /camera/realsense2_camera/rosbag_filename: 
 * /camera/realsense2_camera/serial_no: 
 * /camera/realsense2_camera/stereo_module/exposure/1: 7500
 * /camera/realsense2_camera/stereo_module/exposure/2: 1
 * /camera/realsense2_camera/stereo_module/gain/1: 16
 * /camera/realsense2_camera/stereo_module/gain/2: 16
 * /camera/realsense2_camera/tf_publish_rate: 0.0
 * /camera/realsense2_camera/topic_odom_in: odom_in
 * /camera/realsense2_camera/unite_imu_method: 
 * /camera/realsense2_camera/usb_port_id: 
 * /rosdistro: melodic
 * /rosversion: 1.14.11

NODES
  /camera/
    realsense2_camera (nodelet/nodelet)
    realsense2_camera_manager (nodelet/nodelet)

auto-starting new master
process[master]: started with pid [8374]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 5d3e02c4-da62-11eb-9405-00e04c4bb6c5
process[rosout-1]: started with pid [8387]
started core service [/rosout]
process[camera/realsense2_camera_manager-2]: started with pid [8390]
process[camera/realsense2_camera-3]: started with pid [8391]
[ INFO] [1625140161.369373577]: Initializing nodelet with 4 worker threads.
[ INFO] [1625140162.573053179]: RealSense ROS v2.3.0
[ INFO] [1625140162.573149950]: Built with LibRealSense v2.45.0
[ INFO] [1625140162.573212085]: Running with LibRealSense v2.45.0
[ INFO] [1625140162.658319137]:  
 01/07 07:49:22,752 WARNING [546841690496] (messenger-libusb.cpp:42) control_transfer returned error, index: 300, error: No data available, number: 3d
[ INFO] [1625140162.918016369]: Device with serial number 033322070347 was found.

[ INFO] [1625140162.918121056]: Device with physical ID 2-1.2-3 was found.
[ INFO] [1625140162.918181317]: Device with name Intel RealSense D435I was found.
[ INFO] [1625140162.919603498]: Device with port number 2-1.2 was found.
[ INFO] [1625140162.919696623]: Device USB type: 3.2
[ INFO] [1625140162.919770998]: Resetting device...
[ INFO] [1625140168.967520676]:  
[ INFO] [1625140169.167203438]: Device with serial number 033322070347 was found.

[ INFO] [1625140169.167350000]: Device with physical ID 2-1.2-4 was found.
[ INFO] [1625140169.167404843]: Device with name Intel RealSense D435I was found.
[ INFO] [1625140169.168591715]: Device with port number 2-1.2 was found.
[ INFO] [1625140169.168653538]: Device USB type: 3.2
[ INFO] [1625140169.174691543]: getParameters...
[ INFO] [1625140169.431644633]: setupDevice...
[ INFO] [1625140169.431841559]: JSON file is not provided
[ INFO] [1625140169.431893382]: ROS Node Namespace: camera
[ INFO] [1625140169.431979632]: Device Name: Intel RealSense D435I
[ INFO] [1625140169.432030257]: Device Serial No: 033322070347
[ INFO] [1625140169.432075465]: Device physical port: 2-1.2-4
[ INFO] [1625140169.432122704]: Device FW version: 05.12.10.00
[ INFO] [1625140169.432166402]: Device Product ID: 0x0B3A
[ INFO] [1625140169.432212444]: Enable PointCloud: Off
[ INFO] [1625140169.432253485]: Align Depth: Off
[ INFO] [1625140169.432294318]: Sync Mode: Off
[ INFO] [1625140169.432406141]: Device Sensors: 
[ INFO] [1625140169.465762670]: Stereo Module was found.
[ INFO] [1625140169.511772499]: RGB Camera was found.
[ INFO] [1625140169.512915257]: Motion Module was found.
[ INFO] [1625140169.513056819]: (Confidence, 0) sensor isn't supported by current device! -- Skipping...
[ INFO] [1625140169.513149892]: num_filters: 0
[ INFO] [1625140169.513215464]: Setting Dynamic reconfig parameters.
 01/07 07:49:29,914 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 300, error: No data available, number: 3d
[ INFO] [1625140170.135272844]: Done Setting Dynamic reconfig parameters.
[ INFO] [1625140170.136845809]: depth stream is enabled - width: 848, height: 480, fps: 30, Format: Z16
[ INFO] [1625140170.138799815]: color stream is enabled - width: 1280, height: 720, fps: 30, Format: RGB8
[ INFO] [1625140170.138915127]: setupPublishers...
[ INFO] [1625140170.152543428]: Expected frequency for depth = 30.00000
[ INFO] [1625140170.296427247]: Expected frequency for color = 30.00000
[ INFO] [1625140170.390676083]: setupStreams...
[ INFO] [1625140170.403039438]: insert Depth to Stereo Module
[ INFO] [1625140170.403209073]: insert Color to RGB Camera
 01/07 07:49:30,535 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
[ INFO] [1625140170.585820716]: SELECTED BASE:Depth, 0
[ INFO] [1625140170.630518109]: RealSense Node Is Up!
[ WARN] [1625140170.664565422]: 
 01/07 07:49:30,665 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:30,715 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:30,766 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:30,913 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:30,964 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:31,014 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:31,166 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:31,317 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:31,772 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:49:39,337 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:50:13,444 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:50:30,508 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:50:32,560 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 07:51:00,641 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 07:51:36,724 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 07:51:39,777 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 07:51:50,839 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 07:51:56,893 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 07:52:02,949 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 07:52:07,002 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 07:52:13,058 WARNING [546299687296] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
```

**Comment by MartyG-RealSense** [comment]
Thank you very much.  The versions of the RealSense ROS wrapper and librealsense SDK are correctly matched.

I note your comment that "very few of the topics are being published".  A number of the topics are set to False by default in the **rs_camera** launch file and therefore have to be enabled in order to be published.  The disabled by default topics that are supported on D435i include: infra1 infra2, gyro and accel.  You can enable these topics by including the commands below in your roslaunch instruction:

enable_infra1:=true
enable_infra2:=true
enable_gyro:=true
enable_accel:=true

For example:

**roslaunch realsense2_camera rs_camera.launch enable_infra1:=true enable_infra2:=true enable_gyro:=true enable_accel:=true**

**Comment by MiPlayer123** [comment]
Thanks, I tried that, but I noticed that I am still running into issues. I can see now that the topics are being published, but I cannot see anything when I add these topic to Rviz. 
I ran `roslaunch realsense2_camera rs_camera.launch enable_infra1:=true enable_infra2:=true enable_gyro:=true enable_accel:=true`  and also ran `roslaunch realsense2_camera rs_camera.launch initial_reset:=true enable_infra1:=true enable_infra2:=true enable_gyro:=true enable_accel:=true`

Side question, is there a example launch file that has all these topics enabled and the parameters set properly?
Anyway, here is the output log:

```
SUMMARY
========

PARAMETERS
 * /camera/realsense2_camera/accel_fps: -1
 * /camera/realsense2_camera/accel_frame_id: camera_accel_frame
 * /camera/realsense2_camera/accel_optical_frame_id: camera_accel_opti...
 * /camera/realsense2_camera/align_depth: False
 * /camera/realsense2_camera/aligned_depth_to_color_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/allow_no_texture_points: False
 * /camera/realsense2_camera/base_frame_id: camera_link
 * /camera/realsense2_camera/calib_odom_file: 
 * /camera/realsense2_camera/clip_distance: -2.0
 * /camera/realsense2_camera/color_fps: -1
 * /camera/realsense2_camera/color_frame_id: camera_color_frame
 * /camera/realsense2_camera/color_height: -1
 * /camera/realsense2_camera/color_optical_frame_id: camera_color_opti...
 * /camera/realsense2_camera/color_width: -1
 * /camera/realsense2_camera/confidence_fps: -1
 * /camera/realsense2_camera/confidence_height: -1
 * /camera/realsense2_camera/confidence_width: -1
 * /camera/realsense2_camera/depth_fps: -1
 * /camera/realsense2_camera/depth_frame_id: camera_depth_frame
 * /camera/realsense2_camera/depth_height: -1
 * /camera/realsense2_camera/depth_optical_frame_id: camera_depth_opti...
 * /camera/realsense2_camera/depth_width: -1
 * /camera/realsense2_camera/device_type: 
 * /camera/realsense2_camera/enable_accel: True
 * /camera/realsense2_camera/enable_color: True
 * /camera/realsense2_camera/enable_confidence: True
 * /camera/realsense2_camera/enable_depth: True
 * /camera/realsense2_camera/enable_fisheye1: False
 * /camera/realsense2_camera/enable_fisheye2: False
 * /camera/realsense2_camera/enable_fisheye: False
 * /camera/realsense2_camera/enable_gyro: True
 * /camera/realsense2_camera/enable_infra1: True
 * /camera/realsense2_camera/enable_infra2: True
 * /camera/realsense2_camera/enable_infra: False
 * /camera/realsense2_camera/enable_pointcloud: True
 * /camera/realsense2_camera/enable_pose: False
 * /camera/realsense2_camera/enable_sync: False
 * /camera/realsense2_camera/filters: 
 * /camera/realsense2_camera/fisheye1_frame_id: camera_fisheye1_f...
 * /camera/realsense2_camera/fisheye1_optical_frame_id: camera_fisheye1_o...
 * /camera/realsense2_camera/fisheye2_frame_id: camera_fisheye2_f...
 * /camera/realsense2_camera/fisheye2_optical_frame_id: camera_fisheye2_o...
 * /camera/realsense2_camera/fisheye_fps: -1
 * /camera/realsense2_camera/fisheye_frame_id: camera_fisheye_frame
 * /camera/realsense2_camera/fisheye_height: -1
 * /camera/realsense2_camera/fisheye_optical_frame_id: camera_fisheye_op...
 * /camera/realsense2_camera/fisheye_width: -1
 * /camera/realsense2_camera/gyro_fps: -1
 * /camera/realsense2_camera/gyro_frame_id: camera_gyro_frame
 * /camera/realsense2_camera/gyro_optical_frame_id: camera_gyro_optic...
 * /camera/realsense2_camera/imu_optical_frame_id: camera_imu_optica...
 * /camera/realsense2_camera/infra1_frame_id: camera_infra1_frame
 * /camera/realsense2_camera/infra1_optical_frame_id: camera_infra1_opt...
 * /camera/realsense2_camera/infra2_frame_id: camera_infra2_frame
 * /camera/realsense2_camera/infra2_optical_frame_id: camera_infra2_opt...
 * /camera/realsense2_camera/infra_fps: 30
 * /camera/realsense2_camera/infra_height: 480
 * /camera/realsense2_camera/infra_rgb: False
 * /camera/realsense2_camera/infra_width: 848
 * /camera/realsense2_camera/initial_reset: True
 * /camera/realsense2_camera/json_file_path: 
 * /camera/realsense2_camera/linear_accel_cov: 0.01
 * /camera/realsense2_camera/odom_frame_id: camera_odom_frame
 * /camera/realsense2_camera/ordered_pc: False
 * /camera/realsense2_camera/pointcloud_texture_index: 0
 * /camera/realsense2_camera/pointcloud_texture_stream: RS2_STREAM_COLOR
 * /camera/realsense2_camera/pose_frame_id: camera_pose_frame
 * /camera/realsense2_camera/pose_optical_frame_id: camera_pose_optic...
 * /camera/realsense2_camera/publish_odom_tf: True
 * /camera/realsense2_camera/publish_tf: True
 * /camera/realsense2_camera/rosbag_filename: 
 * /camera/realsense2_camera/serial_no: 
 * /camera/realsense2_camera/stereo_module/exposure/1: 7500
 * /camera/realsense2_camera/stereo_module/exposure/2: 1
 * /camera/realsense2_camera/stereo_module/gain/1: 16
 * /camera/realsense2_camera/stereo_module/gain/2: 16
 * /camera/realsense2_camera/tf_publish_rate: 0.0
 * /camera/realsense2_camera/topic_odom_in: odom_in
 * /camera/realsense2_camera/unite_imu_method: 
 * /camera/realsense2_camera/usb_port_id: 
 * /rosdistro: melodic
 * /rosversion: 1.14.11

NODES
  /camera/
    realsense2_camera (nodelet/nodelet)
    realsense2_camera_manager (nodelet/nodelet)

auto-starting new master
process[master]: started with pid [10402]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 44b93494-daa3-11eb-aeed-00e04c4bb6c5
process[rosout-1]: started with pid [10415]
started core service [/rosout]
process[camera/realsense2_camera_manager-2]: started with pid [10432]
process[camera/realsense2_camera-3]: started with pid [10433]
[ INFO] [1625168036.603386508]: Initializing nodelet with 4 worker threads.
[ INFO] [1625168036.957856228]: RealSense ROS v2.3.0
[ INFO] [1625168036.957956027]: Built with LibRealSense v2.45.0
[ INFO] [1625168036.958029313]: Running with LibRealSense v2.45.0
[ INFO] [1625168037.058753620]:  
 01/07 15:33:57,108 WARNING [547631870336] (messenger-libusb.cpp:42) control_transfer returned error, index: 300, error: No data available, number: 3d
 01/07 15:33:57,170 WARNING [547631870336] (messenger-libusb.cpp:42) control_transfer returned error, index: 300, error: No data available, number: 3d
[ INFO] [1625168037.326263722]: Device with serial number 033322070347 was found.

[ INFO] [1625168037.326376647]: Device with physical ID 2-1.2-5 was found.
[ INFO] [1625168037.326440610]: Device with name Intel RealSense D435I was found.
[ INFO] [1625168037.327970100]: Device with port number 2-1.2 was found.
[ INFO] [1625168037.328085421]: Device USB type: 3.2
[ INFO] [1625168037.328179021]: Resetting device...
[ INFO] [1625168043.375890791]:  
 01/07 15:34:03,458 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 15:34:03,623 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
[ INFO] [1625168043.684876575]: Device with serial number 033322070347 was found.

[ INFO] [1625168043.685062109]: Device with physical ID 2-1.2-6 was found.
[ INFO] [1625168043.685158783]: Device with name Intel RealSense D435I was found.
[ INFO] [1625168043.686335693]: Device with port number 2-1.2 was found.
[ INFO] [1625168043.686419345]: Device USB type: 3.2
[ INFO] [1625168043.692566311]: getParameters...
[ INFO] [1625168043.973695033]: setupDevice...
[ INFO] [1625168043.973795405]: JSON file is not provided
[ INFO] [1625168043.973860514]: ROS Node Namespace: camera
[ INFO] [1625168043.973929685]: Device Name: Intel RealSense D435I
[ INFO] [1625168043.973980418]: Device Serial No: 033322070347
[ INFO] [1625168043.974027661]: Device physical port: 2-1.2-6
[ INFO] [1625168043.974071519]: Device FW version: 05.12.10.00
[ INFO] [1625168043.974115741]: Device Product ID: 0x0B3A
[ INFO] [1625168043.974162567]: Enable PointCloud: On
[ INFO] [1625168043.974207883]: Align Depth: Off
[ INFO] [1625168043.974251063]: Sync Mode: On
[ INFO] [1625168043.974386594]: Device Sensors: 
[ INFO] [1625168044.007633742]: Stereo Module was found.
[ INFO] [1625168044.059042905]: RGB Camera was found.
[ INFO] [1625168044.060156112]: Motion Module was found.
[ INFO] [1625168044.060339250]: (Confidence, 0) sensor isn't supported by current device! -- Skipping...
[ INFO] [1625168044.060434726]: Add Filter: pointcloud
[ INFO] [1625168044.061945202]: num_filters: 1
[ INFO] [1625168044.062049950]: Setting Dynamic reconfig parameters.
 01/07 15:34:04,404 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 300, error: No data available, number: 3d
[ INFO] [1625168044.700218141]: Done Setting Dynamic reconfig parameters.
[ INFO] [1625168044.701603712]: depth stream is enabled - width: 848, height: 480, fps: 30, Format: Z16
[ INFO] [1625168044.702279231]: infra1 stream is enabled - width: 848, height: 480, fps: 30, Format: Y8
[ INFO] [1625168044.702929175]: infra2 stream is enabled - width: 848, height: 480, fps: 30, Format: Y8
[ INFO] [1625168044.706001329]: color stream is enabled - width: 1280, height: 720, fps: 30, Format: RGB8
[ INFO] [1625168044.706254889]: gyro stream is enabled - fps: 200
[ INFO] [1625168044.706353855]: accel stream is enabled - fps: 63
[ INFO] [1625168044.706440268]: setupPublishers...
[ INFO] [1625168044.717565985]: Expected frequency for depth = 30.00000
[ INFO] [1625168044.848903189]: Expected frequency for infra1 = 30.00000
[ INFO] [1625168044.948626859]: Expected frequency for infra2 = 30.00000
[ INFO] [1625168045.047890490]: Expected frequency for color = 30.00000
[ INFO] [1625168045.161150823]: setupStreams...
 01/07 15:34:05,162 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 300, error: No data available, number: 3d
 01/07 15:34:05,225 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 300, error: No data available, number: 3d
[ INFO] [1625168045.286777433]: insert Depth to Stereo Module
[ INFO] [1625168045.286991250]: insert Color to RGB Camera
[ INFO] [1625168045.287085528]: insert Infrared to Stereo Module
[ INFO] [1625168045.287151054]: insert Infrared to Stereo Module
[ INFO] [1625168045.287222049]: insert Gyro to Motion Module
[ INFO] [1625168045.287317941]: insert Accel to Motion Module
 01/07 15:34:05,300 WARNING [547657048448] (ds5-motion.cpp:473) IMU Calibration is not available, default intrinsic and extrinsic will be used.
[ WARN] [1625168045.334371881]: 
[ WARN] [1625168045.336991032]: frame's time domain is HARDWARE_CLOCK. Timestamps may reset periodically.
 01/07 15:34:05,459 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
[ INFO] [1625168045.509717627]: SELECTED BASE:Depth, 0
[ INFO] [1625168045.567584134]: RealSense Node Is Up!
 01/07 15:34:05,583 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 15:34:05,633 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 15:34:05,684 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
[ WARN] [1625168045.701075439]: Hardware Notification:Motion Module failure,1.62517e+12,Error,Hardware Error
 01/07 15:34:05,734 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 15:34:05,786 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: No data available, number: 61
 01/07 15:34:06,101 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 15:34:06,153 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 15:34:06,204 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 15:34:06,456 WARNING [547356643712] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 01/07 15:34:27,059 ERROR [546400366976] (synthetic-stream.cpp:48) Exception was thrown during user processing callback: Out of frame resources!
 01/07 15:34:27,094 ERROR [546400366976] (synthetic-stream.cpp:48) Exception was thrown during user processing callback: Out of frame resources!
 01/07 15:34:27,126 ERROR [546400366976] (synthetic-stream.cpp:48) Exception was thrown during user processing callback: Out of frame resources!
 01/07 15:34:27,161 ERROR [546400366976] (synthetic-stream.cpp:48) Exception was thrown during user processing callback: Out of frame resources!
 01/07 15:34:27,193 ERROR [546400366976] (synthetic-stream.cpp:48) Exception was thrown during user processing callback: Out of frame resources!
```

**Comment by MartyG-RealSense** [comment]
Are the **control_transfer returned** messages being generated continuously or do they stop after the ones shown in the log above, please?

Also, which topics are you adding to RViz?  Could you provide an image of your RViz screen if possible?

**rs_aligned_depth.launch** has depth, color, infra and IMU stream types all enabled as True.

https://github.com/IntelRealSense/realsense-ros/blob/development/realsense2_camera/launch/rs_aligned_depth.launch

**Comment by MiPlayer123** [comment]
The control_transfer returned messages stop momentarily after what you see in the output log, but it keeps coming slowly after that. For some reason, I get this error `Hardware Notification:Motion Module failure,1.62524e+12,Error,Hardware Error` even when I reset the camera. 
When I ran `rs_aligned_depth.launch`, i get this error repeatedly:
```
WARNING [547046277504] (messenger-libusb.cpp:42) control_transfer returned error, index: 768, error: Resource temporarily unavailable, number: 11
 02/07 10:57:11,429 ERROR [546039644544] (synthetic-stream.cpp:48) Exception was thrown during user processing callback: Out of frame resources!
 02/07 10:57:11,732 ERROR [546039644544] (synthetic-stream.cpp:48) Exception was thrown during user processing callback: Out of frame resources!
```
with the last error showing up repeatedly.

Here is an image from Rviz.
![image](https://user-images.githubusercontent.com/31391142/124294358-92513300-db25-11eb-993a-b3fa508a1f5d.png)
The Images show up properly, but I am not sure I I should be seeing points from the DepthCloud or not on the grid. The IMUs are not working though. Here are all the topics I see:
![image](https://user-images.githubusercontent.com/31391142/124294576-dba18280-db25-11eb-878e-7d55ce51c705.png)
I am not seeing an Odom topic, am I supposed to?

Thank you for your help so far.

**Comment by MartyG-RealSense** [comment]
If the **control_transfer returned** warning appears only once or twice per minute after launch then it is safe to ignore it.

**Out of frame resources** could suggest that the USB port is being over-burdened by the rate at which incoming data is arriving and the 'frame queue' is being jammed up, preventing new frames that are arriving from being processed.

You could try a simple custom launch where only depth (no color) will be published at a 640x480 resolution and 30 FPS to test how your computer copes with incoming data.

**roslaunch realsense2_camera rs_camera.launch depth_width:=640 depth_height:=480 depth_fps:=30**

As far as I am aware, **odom** is a topic that is produced by launch files designed for use with the **RealSense T265 Tracking Camera**. 

A guide that is based on Intel's D435i SLAM tutorial for the RealSense ROS wrapper does suggest a method for accessing an odom topic with a D435i using **RTABMAP** though.

https://shinkansan.github.io/2019-UGRP-DPoom/SLAM.html

In regard to RViz, do you see images if you change Fixed Frame from 'map' to **camera_link**

**Comment by MiPlayer123** [comment]
The command seems to work fine. I still get the control_transfer returned warning, but I get no other errors. However, when  I run
 with the gyro and accel enabled, i get this error: `Hardware Notification:Motion Module failure,1.62524e+12,Error,Hardware Error` I am not sure how to fix it. 

Thanks for the resource. I looked into it, but i get a **failed to meet update rate** error. 

As for RViz, I am not sure what you mean.

**Comment by MartyG-RealSense** [comment]
Do you still experience the motion module failure if you include the command below in your roslaunch instruction:

**unite_imu_method:=linear_interpolation**

Regarding RViz, I mean going to the **Fixed Frame** option of the RViz side-panel and changing its setting from the **map** that it is currently using so that it says **camera_link**, like the image below.

![image](https://user-images.githubusercontent.com/41145062/124306362-322ab300-db5e-11eb-9e47-6385dffa4f56.png)

**Comment by MiPlayer123** [comment]
I still face the motion module failure error, but the IMU now shows up on Rviz. I am also able to see the DepthCloud now. 
![image](https://user-images.githubusercontent.com/31391142/124310160-ca627100-db39-11eb-8f84-dcd168c9583e.png)

I am wondering if you know how to fix **Failed to meet update rate!** when running `roslaunch realsense2_camera opensource_tracking.launch` Everything works, but it is really slow, so I am guessing its because of this error.

**Comment by MartyG-RealSense** [comment]
Do you see an improvement if you set unite_imu_method to 'copy' instead of 'linear_interpolation' by adding the command below to the roslaunch instruction to override the launch file's default linear_interpolation setting?  

**unite_imu_method:=copy**

**Comment by MiPlayer123** [comment]
No, its the same as before. I still get the error, but I see something like the previous image in RViz. Do you think that the camera's IMU might be affected hardware wise? This is a brand new camera though.

**Comment by MartyG-RealSense** [comment]
There have been past reports of the **Failed to meet update rate!** message happening when using the **opensource_tracking.launch** launch file, so I would not strongly suspect that there is a hardware problem in the camera.  The camera certainly can publish data, though at a slow rate in this particular case.

In the RealSense Viewer, the **Motion Module Failure** error is more of an annoyance than a serious problem, as the accel and gyro streams usually activate normally the next time that they are enabled after the error has occurred.  It is more complicated in RealSense ROS though, as you cannot just press a button to re-start the IMU streams after the failure message has occurred.

Intel provide a Python-based tool to calibrate the IMU, though I would not expect IMU calibration to be a cause of this particular issue.  Indeed, a Nano user who had the **Failed to meet update rate!** message in a past RealSense ROS case performed the IMU calibration but it did not resolve the problem.  Their next step was going to be to reduce the IMU frequency to see if it helped, but they did not report the outcome of that test.

https://github.com/IntelRealSense/realsense-ros/issues/1234#issuecomment-643591578

In another case involving this error, a user reported freeze-up when **Failed to meet update rate!** was being generated, similar to your own significant slowdown.

https://github.com/IntelRealSense/realsense-ros/issues/1793

Is the camera moving when the message is occurring or is the camera stationary?

**Comment by MiPlayer123** [comment]
> I would not strongly suspect that there is a hardware problem in the camera. The camera certainly can publish data, though at a slow rate in this particular case.

Oh ok.

> Intel provide a Python-based tool to calibrate the IMU

Do you happen to know which directory the `rs-imu-calibration.py` is because I am not able to locate it.

> Is the camera moving when the message is occurring or is the camera stationary?

The camera is stationary. I am not sure if the slowdown/ freeze up is caused due to the Jetson Nano not having enough processing power or not. The CPU is up near 100% when I launch it and that could potentially cause the **Failed to meet update rate!** error.

**Comment by MartyG-RealSense** [comment]
**rs-imu-calibration** can be found in the librealsense source code folder **librealsense/tools/rs-imu-calibration**.  It can be launched using the command **python rs-imu-calibration.py**

https://github.com/IntelRealSense/librealsense/tree/master/tools/rs-imu-calibration

Did you build the RealSense ROS wrapper from packages using the **Method 1** installation that installs the librealsense SDK and the wrapper together, please?  If you did then you may not have support installed in the SDK for CUDA graphics acceleration.  You can install CUDA support for your Nano by building the SDK and ROS wrapper separately, as described in the link below.

https://github.com/IntelRealSense/realsense-ros/issues/1911#issuecomment-860245792

**Comment by MiPlayer123** [comment]
I tried installing **librealsense** because I have **librealsense2** and I cant find the /tools/ direcotory, but now I get this error:
![image](https://user-images.githubusercontent.com/31391142/125207302-941fa280-e259-11eb-8fa8-a9ef3d91459f.png)

While doing something, i broke the realsense-ros package somehow and I get this error:
```
CMake Error at /opt/ros/melodic/share/catkin/cmake/catkinConfig.cmake:83 (find_package):
  Could not find a package configuration file provided by
  "ddynamic_reconfigure" with any of the following names:

    ddynamic_reconfigureConfig.cmake
    ddynamic_reconfigure-config.cmake
```
I tried installing Cuda, but I am running into issues. I will try that again later.

Looks like I  might have messed something up, which is causing me to run into these issues. Not sure what happened though.

**Comment by MartyG-RealSense** [comment]
The multiple udev rules warning typically occurs if librealsense has been installed with both the package and source-code methods on the same computer.  Deleting the rule in the **etc/udev** version of the folder can correct the problem.  

The **lib/udev** rule should not be deleted, as doing so can break linkages to the librealsense packages.

**Comment by MartyG-RealSense** [comment]
Hi @MiPlayer123  Do you require further assistance with this case, please?  Thanks!

**Comment by MiPlayer123** [comment]
Sorry, I haven't posted anything lately, I got busy. I am getting the normal Jetson Nano (4GB) and I am going to test on that (with CUDA enabled). I will update here if it works or not.

Thank you for your help so far @MartyG-RealSense. It has been really helpful.

**Comment by MartyG-RealSense** [comment]
You are very welcome @MiPlayer123 - please do update here when you have test results with Jetson Nano.  Good luck!

**Comment by MartyG-RealSense** [comment]
Hi @MiPlayer123  Do you have an update about this case that you can provide, please?  Thanks!

**Comment by MiPlayer123** [comment]
Hi @MartyG-RealSense I just go the new jetson and i will update here with any issues in the next couple days

**Comment by MartyG-RealSense** [comment]
Thanks very much @MiPlayer123  :)

**Comment by MartyG-RealSense** [comment]
Hi @MiPlayer123  Do you have an update about this case that you can provide, please?  Thanks!

**Comment by MartyG-RealSense** [comment]
Hi @MiPlayer123  Do you require further assistance with this case, please?  Thanks!

**Comment by MiPlayer123** [comment]
Hi @MartyG-RealSense I was busy before and was not able to test it. I was able to get it to work and I can see the data in rviz. I still get the control transfer error, but its rare and does not affect performance. I have also gotten a T265 tracking camera for reliable VSLAM. I am wondering how I can run both the tracking and depth camera ros nodes. 

I was not able to install with cuda as the documentation was unclear to me. I get a `failed to meet update rate` when running [https://github.com/IntelRealSense/realsense-ros/wiki/SLAM-with-D435i](https://github.com/IntelRealSense/realsense-ros/wiki/SLAM-with-D435i), but this is not a major issue due to the tracking camera.

**Comment by MartyG-RealSense** [comment]
Thanks @MiPlayer123 for the update.  There is little documentation available about the **Failed to meet update rate!** error. When ROS does generate this error though, it sometimes provides this advice: "Try decreasing the rate, limiting sensor output frequency, or limiting the number of sensors".

I am not involved in support of the T265 Tracking Camera model and so would recommend creating a new issue on this forum with 'T265' in the message title if you have T265 questions.  You could though try launching a T265 and D435i together with **rs_d400_and_t265.launch**

**Comment by MartyG-RealSense** [comment]
Case closed due to D435i performance being declared satisfactory and advice provided to create a T265 question as a separate case.
